### ISSUE TITLE
opt: ensure that windowing queries are efficient with partitioned indexes

### ISSUE THREAD

**Issue by rytaft** [cockroachdb/cockroach#56203]
As discussed offline with @RaduBerinde:

When partitioning an index (unique or not), we lose efficiency for an important access pattern: doing "windowing" using queries like `WHERE x > last_x LIMIT 100`. To fix this, we would want to split the scan between regions so that each one can be limited.

To achieve this, we may be able to extend `SplitScanIntoUnionScans` (as described in #55156).

**Comment by rytaft** [comment]
Hey @RaduBerinde,
I think this issue is already fixed by the fix to #55156. For example:
```
demo@127.0.0.1:26257/multi_region_test_db> explain select * from regional_by_row_table WHERE pk > 10 order by pk LIMIT 100;
                               info
-------------------------------------------------------------------
  distribution: full
  vectorized: true

  • limit
  │ count: 100
  │
  └── • sort
      │ order: +pk
      │
      └── • union
          │
          ├── • union
          │   │
          │   ├── • scan
          │   │     estimated row count: 100 (3.3% of the table)
          │   │     table: regional_by_row_table@primary
          │   │     spans: [/'europe-west1'/11 - /'europe-west1']
          │   │     limit: 100
          │   │
          │   └── • scan
          │         estimated row count: 100 (3.3% of the table)
          │         table: regional_by_row_table@primary
          │         spans: [/'us-east1'/11 - /'us-east1']
          │         limit: 100
          │
          └── • scan
                estimated row count: 100 (3.3% of the table)
                table: regional_by_row_table@primary
                spans: [/'us-west1'/11 - /'us-west1']
                limit: 100
```
Do you think we need to do anything else here?

**Comment by RaduBerinde** [comment]
Looks great, thanks for checking!
